src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -297,150 +333,264 @@ static const std::map<cldnn::format::type, dnnl::memory::format_tag> format_map_
+                strides.push_back(1);
+                strides.push_back(padded_dims[1]);
+                break;
+
+            default: {
+                auto pitches = _layout.get_pitches();
+                strides.assign(pitches.begin(), pitches.end());
+                break;
+            }
         }
-        dnnl::memory::desc res(dims, dt, strides);
-        return res;
-    } else {
-        dnnl::memory::format_tag fmt = target_fmt == dnnl::memory::format_tag::undef ? convert_data_format(l.format) : target_fmt;
-        dnnl::memory::desc res(dims, dt, fmt);
-        return res;
+
+        return strides;
     }
+};
+
+// Public API functions using the builder
+dnnl::memory::desc layout_to_memory_desc(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).build();
+}
+
+dnnl::memory::desc layout_to_memory_desc_flatten(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).as_flattened().build();
+}
+
+dnnl::memory::desc layout_to_memory_desc_blocked(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).keep_rank().build();
+}
+
+dnnl::memory::desc layout_to_memory_desc_grouped(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).keep_rank().build();
 }
+
+dnnl::memory::desc layout_to_memory_desc_strides(const cldnn::layout& l, dnnl::memory::format_tag target_fmt) {
+    return MemoryDescriptorBuilder(l, target_fmt).with_strides().build();
+}
+
+dnnl::memory::desc layout_to_memory_desc(cldnn::layout l, bool use_default_format, bool is_output_blocked) {
+    auto md = MemoryDescriptorBuilder(l, dnnl::memory::format_tag::undef);
+
+    if (use_default_format) {
+        md.change_target_fmt(get_default_data_format(l));
+        return md.build();
+    }
+
+    if (is_output_blocked) {
+        md.keep_rank();
+    }
+
+    return md.build();
+}
+
+std::tuple<dnnl::memory::desc, dnnl::memory::desc, dnnl::memory::desc>
+get_conv_memory_descs(cldnn::layout input_layout, cldnn::layout weights_layout, cldnn::layout output_layout, dnnl::memory::format_tag target_fmt) {
+    bool need_blocked = input_layout.format.is_blocked() || output_layout.format.is_blocked();
+    bool is_grouped = format::is_grouped(weights_layout.format);
+
+    auto input_mdb = MemoryDescriptorBuilder(input_layout, target_fmt);
+    auto weights_mdb = MemoryDescriptorBuilder(weights_layout, dnnl::memory::format_tag::any);
+    auto output_mdb = MemoryDescriptorBuilder(output_layout, target_fmt);
+
+    if (need_blocked || is_grouped) {
+        input_mdb.keep_rank();
+        weights_mdb.keep_rank();
+        output_mdb.keep_rank();
+    }
+
+    auto input_desc = input_mdb.build();
+    auto weights_desc = weights_mdb.build();
+    auto output_desc = output_mdb.build();
+
+#ifdef GPU_DEBUG_CONFIG
```

Comment:
We don't need #ifdef block GPU_DEBUG_CONFIG because GPU_DEBUG_TRACE_DETAIL handles it.